aws-auth/src/main/java/software/amazon/awssdk/awsauth/regions/Region.java

```diff
@@ -13,7 +13,7 @@
  * permissions and limitations under the License.
  */
 
-package software.amazon.awssdk.core.regions;
+package software.amazon.awssdk.awsauth.regions;
```

Comment:
Auth and region are related but not necesarrily coupled. Should region related stuff be in a separate module?

Reply:
Yes, we can further split aws-auth module to `region`, `profile`, `credentials`, `signer`, I'll try and see how it looks like, not sure if they have circular dependency right now.

Reply:
profile classes are moved to profiles module via commit https://github.com/aws/aws-sdk-java-v2/pull/475/commits/70825f422e21e65393d91394a052f67daa3e0105. 

As to regions and credential classes, they are tightly coupled now because they share common functionalities to to retrieve regions/creds from local EC2 instances in `HttpCredentialUtils`, `endpointRetry` and other classes, unless we break out those class to a separate module, it's difficult to separate them. If we are going with that route, we might end up having plenty "micro" modules and it could increase our maintenance burden and unlike profiles, regions and creds seem to be always used together.